uberduck_ml_dev/trainer/tacotron2.py

```diff
@@ -186,31 +208,55 @@ def log_training(
                     )
                 ),
             )
-            if self.distributed_run:
-                self.sample_inference(model.module)
-            else:
-                self.sample_inference(model)
+            for speaker_id in self.sample_inference_speaker_ids:
+                if self.distributed_run:
+                    self.sample_inference(
+                        model.module,
+                        self.sample_inference_text,
+                        speaker_id,
+                    )
+                else:
+                    self.sample_inference(
+                        model,
+                        self.sample_inference_text,
+                        speaker_id,
+                    )
 
-    def sample_inference(self, model):
+    def sample_inference(self, model, transcription=None, speaker_id=None):
         if self.rank is not None and self.rank != 0:
             return
         # Generate an audio sample
         with torch.no_grad():
+            if transcription is None:
+                transcription = random_utterance()
+
+            if self.compute_gst:
+                gst_embedding = self.compute_gst([transcription])
+                gst_embedding = torch.FloatTensor(gst_embedding).cuda()
+            else:
+                gst_embedding = None
+
             utterance = torch.LongTensor(
                 text_to_sequence(
-                    random_utterance(),
+                    transcription,
                     self.text_cleaners,
                     p_arpabet=self.p_arpabet,
                     symbol_set=self.symbol_set,
                 )
             )[None].cuda()
-            speaker_id = (
-                choice(self.sample_inference_speaker_ids)
-                if self.sample_inference_speaker_ids
-                else randint(0, self.n_speakers - 1)
-            )
+            if speaker_id is None:
+                speaker_id = (
+                    choice(self.sample_inference_speaker_ids)
+                    if self.sample_inference_speaker_ids
+                    else randint(0, self.n_speakers - 1)
```

Comment:
I think we can remove this if since `sample_inference_speaker_ids` is now always set